server/routerlicious/packages/lambdas/src/nexus/index.ts

```diff
@@ -455,20 +466,39 @@ export function configureWebSocketServices(
 						socket.emit("nack", "", [nackMessage]);
 						return;
 					}
-					contentBatches.forEach((contentBatch) => {
-						const contents = Array.isArray(contentBatch)
-							? contentBatch
-							: [contentBatch];
 
-						for (const content of contents) {
-							const signalMessage: ISignalMessage = {
-								clientId,
-								content,
-							};
+					if (supportedFeaturesMap.get(clientId)?.submit_signals_v2) {
+						for (const signal of contentBatches) {
+							if (isSentSignalMessage(signal)) {
+								const signalMessage: ISignalMessage = {
+									clientId,
+									...signal,
+								};
+
+								const roomId: string =
+									signal.targetClientId !== undefined
+										? getClientSpecificRoomId(signal.targetClientId)
+										: getRoomId(room);
 
-							socket.emitToRoom(getRoomId(room), "signal", signalMessage);
+								socket.emitToRoom(roomId, "signal", signalMessage);
+							}
```

Comment:
Per @jason-ha's comment about whether to enforce "client that supports v2 signals always submits v2 signals," I'm guessing this is the location of concern.

Would it be better to allow v1 signals from a v2 client? Otherwise it seems like a v1 signal from a v2 client will be lost, and I don't see the downside of falling back of just sending the signal as `content` in that case.